src/hdinsightonaks/azext_hdinsightonaks/aaz/latest/hdinsightonaks/_check_name_availability.py

```diff
@@ -0,0 +1,190 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "hdinsightonaks check-name-availability",
+    is_preview=True,
+)
+class CheckNameAvailability(AAZCommand):
+    """Check the availability of the resource name.
+
+    :example: Check if the name clicluster  is available in cluster pool cliclusterpool.
+        az az hdinsightonaks check-name-availability -l westus3 --name cliclusterpool/clicluster --type Microsoft.HDInsight/clusterPools/clusters
```

Comment:
Please fix the examples
```suggestion
        az hdinsightonaks check-name-availability -l westus3 --name cliclusterpool/clicluster --type Microsoft.HDInsight/clusterPools/clusters
```